### PR TITLE
Fp value status

### PR DESCRIPTION
**Pull request by joakim-hove**
Use enum `value::status` to keep track of defaulted / set / ... in FieldProps

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
If still green I will merge this later today.

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Grid/FieldProps.hpp [hunk 1]
```diff
@@ -47,14 +48,14 @@ class FieldProps {
...
-        std::vector<bool> assigned;
+        std::vector<value::status> value_status;
 
 
         FieldData() = default;
 
         FieldData(std::size_t active_size) :
             data(std::vector<T>(active_size)),
-            assigned(std::vector<bool>(active_size, false))
+            value_status({active_size, value::status::uninitialized})
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Grid/FieldProps.hpp]: Does this really do what you want?  The `initializer_list` constructor for `std::vector` assigns elements.  I'm actually surprised this compiles at all.  For safety you should write
```C++
value_status(active_size, value::status::uninitialized)
```
here

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Grid/FieldProps.hpp]: Thanks - I have updated; I guess the original version might have created an additional copy and assigned that?

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**